src/test/java/com/atlassian/jira/cloud/jenkins/listeners/AutoBuildsAndDeploymentsTest.java

```diff
@@ -79,37 +85,48 @@ public void setUp() throws Exception {
         givenJiraAcceptsDeployments();
         givenAutoBuildsRegex(null);
         JiraSenderFactory.setInstance(mockSenderFactory);
+        JiraCloudPluginConfig.get().setDebugLogging(Boolean.TRUE);
     }
 
     private void assertListenersRegistered() {
 
-        // Adding a custom JenkinsPipelineListener so we can inject mocks.
-        // The default listener instance will also still be registered (I didn't find a way to
-        // unregister it),
-        // but it will not send any requests to Jira because it won't find any issue keys.
+        Optional<RunListener> existingListener = jenkins.getInstance()
+                .getExtensionList(RunListener.class)
+                .stream()
+                .filter(listener -> listener instanceof JenkinsPipelineRunListener)
+                .findFirst();
+
+        if (existingListener.isPresent()) {
```

Comment:
Jenkins instantiates and adds a `JenkinsPipelineRunListener` by itself. That listener instantiates an issue key extractor. Since the issue key extractor now looks for issue keys in the `CHANGE_BRANCH` env var, and Jenkins sets this env var in the Jenkins build of this project, the CI build behaved differently from the local build:

- In the local build, the env var is not set and it doesn't find any issue keys. 
- In the CI build, the env var is set, and it finds the issue key in it, which leads to more build events sent to Jira than we expect.

So, I just remove the default listener here before adding a custom listener with our mock issue key extractor, so we can control the issue keys it finds. Took a couple attempts to get this right because of the CI turnaround :(